slick.grid.js

```diff
@@ -2814,7 +2814,10 @@ if (typeof Slick === "undefined") {
         invalidateRow(getDataLength());
       }
 
+      var originalOptions = $.extend(true, {}, options);
       options = $.extend(options, args);
+      trigger(self.onSetOptions, { "optionsBefore": originalOptions, "optionsAfter": options });
```

Comment:
@6pac 
I added a new `onSetOptions` which I then use the Grid Menu plugin to know when users dynamically changes from a regular grid to a frozen grid (or vice versa). Are you ok with this new `onSetOptions` event with `optionsBefore` and `optionsAfter` or if you prefer different naming?

Reply:
No problem. I have toyed with the idea of switching to beforeXXX and afterXXX events (here, for example, before the options are set, and then after), but that might be a breaking change best left to  v3.x.  Here, it wouldn't be needed anyway.